synapse/storage/database.py

```diff
@@ -1755,10 +1755,14 @@ def simple_delete_txn(
         Returns:
             The number of deleted rows.
         """
-        sql = "DELETE FROM %s WHERE %s" % (
-            table,
-            " AND ".join("%s = ?" % (k,) for k in keyvalues),
-        )
+
+        if keyvalues:
+            sql = "DELETE FROM %s WHERE %s" % (
+                table,
+                " AND ".join("%s = ?" % (k,) for k in keyvalues),
+            )
+        else:
+            sql = "DELETE FROM %s" % (table,)
```

Comment:
This seems a bit foot-gunny. Since it is only used in a test maybe we can just inline the SQL there?

Reply:
Yeah, I guess you're right. Especially since postgres doesn't do it efficiently (you need to use `TRUNCATE`). We do correctly handle empty `keyvalues` for other functions, but they're not really footguns